tapgarden/batch.go

```diff
@@ -125,8 +145,10 @@ func (m *MintingBatch) MintingOutputKey() (*btcec.PublicKey, []byte, error) {
 
 // genesisScript returns the script that should be placed in the minting output
 // within the genesis transaction.
-func (m *MintingBatch) genesisScript() ([]byte, error) {
-	mintingOutputKey, _, err := m.MintingOutputKey()
+func (m *MintingBatch) genesisScript(
+	sibling *commitment.TapscriptPreimage) ([]byte, error) {
```

Comment:
Similarly, we should already have this be bound to `MintingBatch` assuming we've read both from disk (or was created in memory).

Reply:
Ah, tho I guess as is, we'll read from the tree store, then supplement the tap sibling read as needed.